exercises/diffie-hellman/tests/diffie-hellman.rs

```diff
@@ -0,0 +1,58 @@
+extern crate diffie_hellman;
+
+use diffie_hellman::*;
+
+#[test]
+fn test_private_key_in_range_key() {
+    let primes: Vec<u64> = vec![5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 773, 9091,
+                                38321, 6131329];
+    let private_keys: Vec<u64> = primes.iter().map(|x| private_key(*x)).collect();
+
+    for i in 0..primes.len() {
+        assert!(1 < private_keys[i] && private_keys[i] < primes[i]);
+    }
+}
+
+#[test]
+#[ignore]
+fn test_public_key_correct() {
+    let p: u64 = 23;
+    let g: u64 = 5;
+
+    let private_key: u64 = 6;
+    let expected: u64 = 7;
+
+    assert_eq!(public_key(p, g, private_key), expected);
+}
+
+#[test]
+#[ignore]
+fn test_secret_key_correct() {
+    let p: u64 = 92233987;
+    let g: u64 = 5371081;
+
+    let private_key = 75233987;
+    let public_key = public_key(p, g, private_key);
+    let secret = secret(p, public_key, private_key);
```

Comment:
so, to confirm, this is showing what happens when someone effectively performs a DH key exchange with themselves, right? Since they're using their own public key and their own private key.

It does not seem to me like this is reflective of a likely real-world usage, but as you know since I am ignorant of DH, there could simply be a use that I am not aware of.

Reply:
In the last test ```test_changed_secret_key()``` they exchange their public keys.

![grafik](https://user-images.githubusercontent.com/17926729/33375918-efa1595a-d50b-11e7-83e0-8ea31ef55bdd.png)

You see that A decrypts Bs public key with As private key and p. 
B decrypts As public key with Bs private key and p. 

If successful, they share the same secret key/session key.
From the [wiki](https://en.wikipedia.org/wiki/Diffie%E2%80%93Hellman_key_exchange): 
```Now s is the shared secret key and it is known to both Alice and Bob, but not to Eve.```